### ISSUE TITLE
Incorrectly warning that there are duplicate definitions with the same name

### ISSUE THREAD

**Issue by Lapz** [intellij-rust/intellij-rust#2821]
## Environment
* **Intellij-Rust plugin version:** 0.2.0.2104-182
* **Rust toolchain version:** 1.27.0
* **IDE name and version:** IntelliJ IDEA 2018.2.2 (Community Edition),Build #IC-182.4129.33
* **Operating system:** OSX 10.12.6

## Problem description

IntelliJ rust incorrectly says that there are duplicate definitions with the same name citing error [[0201]](https://doc.rust-lang.org/error-index.html#E0201)

## Steps to reproduce
```rust 
pub struct Foo {
    bar: i32,
    #[cfg(feature = "debug")]
    foo: i32,
}

impl Foo {
    fn new() -> Foo {
        Foo { bar: 0 }
    }

    #[cfg(feature = "debug")]
    fn new() -> Foo {
        Foo { bar: 0, foo: 1 }
    }
}
```

**Comment by Lapz** [comment]
It should be #[not(cfg(feature = "debug"))]  but the error should only be visible if the feature debug is enabled

**Comment by alexander-irbis** [comment]
There is no such option as to disable or enable a feature for project.
The IDE does not know which options will be used to compile in each case, it is forced to analyze all the options. Otherwise, how will autocompletion and navigation work?.
In addition, the IDE's job is to determine that with some options the code will not be correct when the programmer does not notice it.
Perhaps, it would be nice to be able limit the set of options for building a project, but this is a separate big feature and it is worth creating an appropriate request for it.

**Comment by kumbayo** [comment]
With the current development version of #2166 no error is reported anymore for the bugfixed version of this code.
In that branch IntelliJ knows which feature options are enabled and which ones are disabled.
But that merge request is still unfinished/under development and it is not clear how good this approach is.
